File: dailybyte/02_valid_palindrome_w_removal/valid.py
def validPalindrome(str):
    left = 0
    right = len(str) - 1
    while left < right:
        if str[left] != str[right]:
            return isPalindrome(str, left + 1, right) or isPalindrome(str, left, right - 1)
        left += 1
        right -= 1
    return True

def isPalindrome(str, left, right):
    while left < right:
        if str[left] != str[right]:
            return False
        left += 1
        right -= 1
    return True

print(validPalindrome("abcba"))
print(validPalindrome("foobof"))
print(validPalindrome("abccab"))

File: dailybyte/02_valid_palindrome_w_removal/valid.rb
def validPalindrome(str)
    i = 0
    j = str.length - 1
    while i < j
        if str[i] != str[j]
        return is_palindrome(str, i + 1, j) || is_palindrome(str, i, j - 1)
        end
        i += 1
        j -= 1
    end
    true
    end

def is_palindrome(str, i, j)
    while i < j
        return false if str[i] != str[j]
        i += 1
        j -= 1
    end
    true
end

puts validPalindrome("abcba") # true
puts validPalindrome("foobof") # true
puts validPalindrome("abccab") # false